src/app/rpmostree-db-builtin-diff.cxx

```diff
@@ -162,6 +162,9 @@ rpmostree_db_builtin_diff (int argc, char **argv,
       return FALSE;
     }
 
+  if (!opt_format)
+    opt_format = g_strdup ("block");
```

Comment:
Doesn't this technically leak? I think what we actually need instead is just

```
static const char *opt_format = "block";
```

? (And similarly for `opt_sysroot` in `start-daemon.cxx`.)

Reply:
All of our current `G_OPTION_TYPE_STRING` values leak.  The docs aren't clear about this but it *has* to allocate in some cases.  See this code: https://gitlab.gnome.org/GNOME/glib/-/blob/d48e22e4640f0c8d442aa24bf30396969443e1a6/glib/goption.c#L1286

So while this looks like a new leak, we've always been leaking on the other path.

In any case these are tiny values.  Ideally of course we'd have valgrind as part of our CI, we should do that but the goal is to get to Rust...

Reply:
Hmm, the code you linked to seems to keep ownership of the data AFAICT. (It looks like it allocates another `Change` associated with the `GOptionContext`, which is freed here when the latter is freed: https://gitlab.gnome.org/GNOME/glib/-/blob/d48e22e4640f0c8d442aa24bf30396969443e1a6/glib/goption.c#L1761).

> In any case these are tiny values. Ideally of course we'd have valgrind as part of our CI, we should do that but the goal is to get to Rust...

Yeah, I'm 100% for enabling Rust migration. That said, it's not going to happen overnight, and there's going to be a lot of C/C++ code for a while. It will still need to be maintained for some time and have bugs fixed. (Even if we could, I'd be hesitant to just rewrite everything to Rust in e.g. one release and have to face all the regressions at the same time -- it also leaves less opportunity for learning from the experience.)

Anyway, circling back to the specific code here... seems like `static const char*` does squash that warning in a quick local test. I'm good doing that as a follow-up if you prefer (feel free to slap on the `lgtm` label!).

Reply:
This is messy, see the docs here:
https://gitlab.gnome.org/GNOME/glib/-/blob/d48e22e4640f0c8d442aa24bf30396969443e1a6/glib/goption.c#L387
as well as here:
https://developer.gnome.org/glib/2.66/glib-Commandline-option-parser.html#GOptionEntry

From the latter:
> If arg type is G_OPTION_ARG_STRING or G_OPTION_ARG_FILENAME, the location will contain a newly allocated string if the option was given. That string needs to be freed by the callee using g_free().